stubs/pyinstaller/PyInstaller/building/splash.pyi

```diff
@@ -7,21 +7,23 @@ splash_requirements: list[str]
 # Referenced in https://pyinstaller.org/en/stable/spec-files.html#example-merge-spec-file
 # Not to be imported during runtime, but is the type reference for spec files which are executed as python code
 class Splash(Target):
-    image_file: str
-    full_tk: Incomplete
-    name: Incomplete
-    script_name: Incomplete
-    minify_script: Incomplete
-    max_img_size: Incomplete
-    text_pos: Incomplete
-    text_size: Incomplete
-    text_font: Incomplete
-    text_color: Incomplete
-    text_default: Incomplete
-    always_on_top: Incomplete
-    uses_tkinter: Incomplete
-    script: Incomplete
-    splash_requirements: Incomplete
+    image_file: StrPath
+    full_tk: bool
+    tcl_lib: str
+    tk_lib: str
+    name: StrPath
```

Comment:
In `__init__, `name` is a `str`.

To change it to a `StrPath`, we should validate that all usages of `Splash.name` work with either a `str` or `PathLike[str]`. Here's 1 line I found where this doesn't work with the current type definitions: https://github.com/pyinstaller/pyinstaller/blob/5e69425f3ea4ecea0226ca47f64eba185070440a/PyInstaller/building/api.py#L545

```suggestion
    name: str
```